src/utils/Api.js

```diff
@@ -0,0 +1,105 @@
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  getAppInfo() {
+    return Promise.all([this.getInitialCards(), this.getUserInfo()]);
+  }
+
+
+  _processResponse(res){
+    if(res.ok){
+      return res.json();
+    }
+    Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    })
+    .then(this._processResponse)
+  }
+
+  // Implement POST /cards
+
+  addNewCard({ name, link }) {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({
+        name,
+        link,
+      }),
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    });
```

Comment:
Please, reuse `.then(this._processResponse)` in every request not to duplicate the `ok` checking